### ISSUE TITLE
[LSP] Deleting imported symbol doesn't remove it from ImportScope

### ISSUE THREAD

**Issue by RealTYPICAL** [eclipse/xtext-core#528]
This issue was introduced in 2.13 and worked fine in 2.12.

A small example can be found at the following repo:

https://github.com/RealTYPICAL/MyDSL_Imports

Instructions to replicate are as follows:

1. Start the environment as specified in the README.md in the repo.
2. Delete the declaration from something.mydsl.
3. Verify that tooley.mydsl doesn't have error for non existent declaration.
4. Restart VSCode (ctrl + shift + p: reload window).
5. Verify that non existent declaration is now not found.

DefaultResourceDescriptionStrategy.createEObjectDescriptions(...) correctly doesn't create an object description for the declaration after it is deleted however it seems as though the ImportScope is never forgetting symbols that are indexed upon initialisation. If the declaration doesn't exist when initialising then behaviour works as expected. (It recognises when the symbol does and doesn't exist as appropriate.)

I would expect that just because declarations are available during initialisation, for the declarations not to be available after they are removed.

Maybe there is a mistake in my working.

Are there any particular classes that I should look at to resolve this issue?

Please don't hesitate to ask if you require more feedback.

**Comment by cdietrich** [comment]
hi i could not reproduce this with my helloworld example. unfortunately i dont get your example to build (mwe fails)

**Comment by cdietrich** [comment]
i could not reproduce it with your example when starting a build with your grammar manually ( wwith and without using xtype)

**Comment by RealTYPICAL** [comment]
I just verified that my project works by checking it out and building it myself. Did you use some other IDE or some strange steps to try and get the project working?

Which OS are you using?

I am considering providing a video of the behaviour as verifying the project is working proved that the issue is there for me.

**Comment by cdietrich** [comment]
=> did you freashly clone? you dont have .project files checked in nor manifests?
i dont use ide. just use command line (ide import creates .project files)
=> potentially we screw up the uri conversion once more and its not working windows

**Comment by RealTYPICAL** [comment]
Yes. I freshly cloned.
Okay... I will try on another computer first before getting back to you.
Possibly some system specific setting that I've overlooked.

Thanks for your initial support.

**Comment by cdietrich** [comment]
well the uri calculation stuff was changed about 10 times back and forth and sidewards and maybe it still is broken on windows. i used my own runner and client extension so that i can easily debug

**Comment by cdietrich** [comment]
here is my socket based extension: https://github.com/itemis/xtext-languageserver-example/tree/master/vscode-extension
and the starter: https://github.com/itemis/xtext-languageserver-example/blob/master/org.xtext.example.mydsl.ide/src/org/xtext/example/mydsl/ide/RunServer.java

**Comment by cdietrich** [comment]
`[file:///c%3A/Users/dietrich/git/MyDSL_Imports/mydsl/workspace/src/something.mydsl]` looks very bad as uri

**Comment by kittaakos** [comment]
> looks very bad as uri

Why do you think it looks bad?

**Comment by cdietrich** [comment]
cause at some point the uri back and forth calculation ends up with an uri without that `c%3A`

**Comment by cdietrich** [comment]
e.g. the index state contains

`{file:///c:/Users/dietrich/git/MyDSL_Imports/mydsl/workspace/src/something.mydsl=org.eclipse.xtext.resource.persistence.SerializableResourceDescription@6e1d4ee4, file:///c%3A/Users/dietrich/git/MyDSL_Imports/mydsl/workspace/src/something.mydsl=org.eclipse.xtext.resource.persistence.SerializableResourceDescription@248b29b3, file:///c:/Users/dietrich/git/MyDSL_Imports/mydsl/workspace/src/tooley.mydsl=org.eclipse.xtext.resource.persistence.SerializableResourceDescription@3f965b79}`

**Comment by kittaakos** [comment]
That is inconsistent. Is there a thing I assist you with issue?

**Comment by kittaakos** [comment]
I still believe, this is the preferred format on Windows: `file:///c%3A/Users/userName`.

**Comment by cdietrich** [comment]
hmmm have never seen this in my emf experience back then when reg. using windows.
yes analyzing on a windows vm is very painful.

i assume that when we do the inital build we create uris for the files in the directory that look different from the ones we use at didChange

**Comment by kittaakos** [comment]
> hmmm have never seen this in my emf experience

That is correct. But the LSP uses that format.

@akosyakov, could you please confirm this? Thank you!

**Comment by cdietrich** [comment]
yes. nut at some point we cannot deal with that. the question is which one

**Comment by RealTYPICAL** [comment]
Just FYI. I committed the .project files to the repository just in case you still need the hello world project.

**Comment by cdietrich** [comment]
e.g. ProjectManager.doInitialBuild creates uris with file:///C:/.....

**Comment by cdietrich** [comment]
i wonder if we should do something like

```
class UriExtensions {

	/**
	 * returns a URI with empty authority, if absent and has file scheme.
	 */
	def URI toUri(String stringUri) {
		val uri = URI.createURI(URLDecoder.decode(stringUri, "UTF-8"));
		return uri.withEmptyAuthority
	}
```

**Comment by cdietrich** [comment]
@RealTYPICAL could you do some tests with Xtext `2.14.0-SNAPSHOT` consumed from `http://services.typefox.io/open-source/jenkins/job/xtext-core/job/cd_issue528/lastSuccessfulBuild/artifact/build/maven-repository/`

**Comment by RealTYPICAL** [comment]
It's working properly. Thanks very much for your support.

**Comment by cdietrich** [comment]
Fixed in 2.14
